### PR TITLE
best-practices: add timeout options for Java best practices

### PR DESCRIPTION
**Pull request by jackysp**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lysu — APPROVED** [review]

File: best-practices/java-app-best-practices.md [hunk 1]
```diff
@@ -182,6 +182,12 @@ enableQueryTimeouts = false
 
 配置后查看监控，可以看到多余语句减少。
 
+#### 超时参数
+
+TiDB 提供两个与 MySQL 兼容的超时控制参数，`wait_timeout` 和 `max_execution_time`，这两个参数分别控制了连接的空闲超时时间和连接的 SQL 执行超时时间，也就是控制了连接最长闲多久和最长忙多久。它们的默认值都是 0，也就是默认允许连接无限闲置以及无限忙碌（一个 SQL 执行无限的长的时间）。
```
Comment (TomShawn) [inline best-practices/java-app-best-practices.md:187]:
```suggestion
TiDB 提供两个与 MySQL 兼容的超时控制参数，`wait_timeout` 和 `max_execution_time`。这两个参数分别控制与 Java 应用连接的空闲超时时间和连接中 SQL 执行的超时时间，即控制 TiDB 与 Java 应用的连接最长闲多久和最长忙多久。这两个参数的默认值都是 `0`，即默认允许连接无限闲置以及无限忙碌（一个 SQL 执行无限的长的时间）。
```

File: best-practices/java-app-best-practices.md [hunk 2]
```diff
@@ -182,6 +182,12 @@ enableQueryTimeouts = false
...
+
+但是，实际生产中，空闲连接和一直无限执行的 SQL 对数据库和应用都有不好的影响，可以通过在应用连接字符串中配置它们来避免空闲连接和执行时间过长的 SQL。比如，设置 `sessionVariables=wait_timeout=3600`（1 小时）和 `sessionVariables=max_execution_time=300000`（5 分钟）。
```
Comment (TomShawn) [inline best-practices/java-app-best-practices.md:189]:
```suggestion
但在实际生产环境中，空闲连接和一直无限执行的 SQL 对数据库和应用都有不好的影响。你可以通过在应用的连接字符串中配置这两个参数来避免空闲连接和执行时间过长的 SQL 语句。例如，设置 `sessionVariables=wait_timeout=3600`（1 小时）和 `sessionVariables=max_execution_time=300000`（5 分钟）。
```

**Comment by jackysp** [comment]
> minor comments~

Done. Thanks!

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**